### PR TITLE
feat: add a flag to mark projects as template

### PR DESCRIPTION
**Pull request by m-alisafaee**
* Adds a `template` flag to `projects` table to mark a project as a template project
* Adds a filter to get writable copies when getting list of projects that are copied from a specific project

### CODE REVIEW

File: components/renku_data_services/migrations/versions/08ac2714e8e2_add_template_flag_to_projects.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# revision identifiers, used by Alembic.
+revision = "08ac2714e8e2"
+down_revision = "ea52d750e389"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.add_column("projects", sa.Column("template", sa.Boolean(), default=False, nullable=False), schema="projects")
```
Comment (m-alisafaee) [inline components/renku_data_services/migrations/versions/08ac2714e8e2_add_template_flag_to_projects.py:21]: Suggestions for a better name are welcome! We can name it `is_template` which shows the intention better.

Comment (Panaetius) [reply components/renku_data_services/migrations/versions/08ac2714e8e2_add_template_flag_to_projects.py:21]: `is_template` does sound better to me

File: components/renku_data_services/project/api.spec.yaml [hunk 2]
```diff
@@ -193,6 +193,13 @@ paths:
           required: true
           schema:
             $ref: "#/components/schemas/Ulid"
+        - in: query
+          description: When true, only return projects that user have write access to them
```
Comment (Panaetius) [inline components/renku_data_services/project/api.spec.yaml:197]:
```suggestion
          description: When true, only return projects that the user has write access to
```

File: components/renku_data_services/project/models.py [hunk 3]
```diff
@@ -63,6 +64,8 @@ class ProjectPatch:
     description: str | None
     keywords: list[str] | None
     documentation: str | None
+    template_id: str | None
```
Comment (Panaetius) [inline components/renku_data_services/project/models.py:67]: suggestion: I wonder if we can type it as `Literal[""]|None` to make it clearer what can/can't be set.

**Review by Panaetius — CHANGES_REQUESTED** [review]

File: components/renku_data_services/project/models.py (hunk 3 above)
Comment (m-alisafaee) [reply components/renku_data_services/project/models.py]: Good point! I've tried to do the same in the API spec but the generated model has an issue. I've filed a bug report for it: https://github.com/koxudaxi/datamodel-code-generator/issues/2191

**Review by Panaetius — APPROVED** [review]

**Review by Panaetius — APPROVED** [review]

**Pull request merged**